packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_list.dart

```diff
@@ -82,11 +82,18 @@ class _ListControlPane extends StatelessWidget {
               icon: iconToTakeSnapshot,
               size: defaultIconSize,
               tooltip: 'Take heap snapshot for the selected isolate',
-              onPressed: controller.isTakingSnapshot.value
+              onPressed: controller.isAddingSnapshot.value
                   ? null
                   : () => unawaited(_takeSnapshot(context)),
             ),
             const SizedBox(width: densePadding),
+            ToolbarAction(
+              icon: Icons.file_upload,
+              tooltip: 'Import snapshots.',
+              onPressed: controller.isAddingSnapshot.value
```

Comment:
this may be too aggressive. What if I upload single snapshot file, then want to upload a second? The upload button would be blocked because we are loading the first.